core/chain_manager.go

```diff
@@ -487,15 +546,37 @@ const (
 	SideStatTy
 )
 
+/*
+// WriteHeader inserts a potentially new header into the database, updating the
+// head-header-hash index too if greater than the previous.
+func (self *ChainManager) WriteHeader(db common.Database, header *types.Header) error {
+	self.wg.Add(1)
+	defer self.wg.Done()
+
+	// Store the header into the database
+	if err := WriteHeader(self.chainDb, header); err != nil {
+		glog.Fatalf("failed to write header: %v", err)
+		return err
+	}
+	// If the header's higher than our previous head, update
+	if header.
+}*/
+
 // WriteBlock writes the block to the chain (or pending queue)
-func (self *ChainManager) WriteBlock(block *types.Block, queued bool) (status writeStatus, err error) {
+func (self *ChainManager) WriteBlock(block *types.Block) (status writeStatus, err error) {
 	self.wg.Add(1)
 	defer self.wg.Done()
 
+	// Calculate the total difficulty of the block
+	td := new(big.Int).Add(block.Difficulty(), self.GetTd(block.ParentHash()))
```

Comment:
The genesis blocks are really written via the direct methods (e.g. `WriteBlock(db, block)`), as the chain manager does not exist when initializing a pristine database. Later on blocks should only be pushed into the chain if the parent exists, so all code calling this function already verifies this.

A more appropriate question may be whether we need this function public or not. It is a relic from an older design, and currently all clients add blocks via `chain_manager.InsertChain`, so we could make this private and then any potential issues are circumvented. Will need to look into all the uses of this method.